NERDS. Сделал по заданию 5 главы 1 курса Исправил замечания. Сделал базовую стилизацию страницы. Проверил. Прогнал CSS на валидацию -все работает.

Nomad48JMS opened the pull request:
Внес правки.

Nomad48JMS committed:
document structure

добавил структурные папки и файл

Nomad48JMS committed:
Summary

Nomad48JMS committed:
Summary

Сделал част разметки главной страницы и добавил каталог.

Nomad48JMS committed:
Update

Сделал разметку главной Nerds
Разметил хедер и футер каталога

Nomad48JMS committed:
Update

Доделал разметку каталога

Nomad48JMS committed:
Validation

Прогнал 2 страницы через валидатор и исправил недочёты.

Nomad48JMS committed:
Exported images

Экспортировал графику. CSS по ходу дела буду смотреть.

Nomad48JMS committed:
Exporting images and connecting content images.

Экспорт изображений и подключение контентных изображений.

Nomad48JMS committed:
Update

Добавил ТЗ в редми

Nomad48JMS committed:
Deleted old images. Exported again. Made the names of the images readable.

Удалил старые изображения. Экспортировал заново. Сделал читабельными названия изображений.

Nomad48JMS committed:
Merge remote-tracking branch 'upstream/master'

Nomad48JMS committed:
I resave the incorrectly exported files again in the correct format. Changed the connection paths in the code. Added nulling and visually-hidden styles and then commented them out.

Неправильно экспортированные файлы сохранил снова в нужном формате. Изменил пути подключения в коде. Добавил обнуляющие стили и стили класса visually-hidden/ А потом их закомментировал.

Nomad48JMS committed:
Так по мелочи. Переехал на VSCode)

Так по мелочи. Переехал на VSCode)

Nomad48JMS committed:
Made on assignment 4 chapters of 1 course.

Причесал код. Сделал по заданию 4 главы 1 курса.
Проверь пожалуйста. Прогнал на валидаторе все работает. Спасибо!:)

Nomad48JMS committed:
Moved explanatory comments to the top of the page. It's more convenient this way.

Перенес поясняющие комментарии вверх страниц. Так удобней.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+<!-- Сделал по заданию 4 главы 1 курса -->
```
Общие замечания:

1. Пожалуйста, относись серьезней к названию коммитов. Представь, что ты работаешь в команде.
"Так, по мелочи" - что это? 
"Переехал на VSCode" - какое это вообще имеет отношение к коду?!

2. Добавь файл .gitignore с текстом /layout

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+<html lang="ru">
+  <head>
+    <meta charset="utf-8">
+    <title>Студия Nerds каталог</title>
```
Все так же недовольна твоими заголовками. Ставь хотя бы тире между названием сайта и страницы.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+  <body>
+    <header class="main-header">
+      <nav class="main-navigation">
+        <img class="logo-main" src="img/nerds-logo.png" width="160" height="65" alt="Логотип студии Nerds">
```
Логотип обычно является ссылкой на главную. Это устоявшаяся практика. К тому же это прямо указано в ТЗ:
3.7. Логотип — это ссылка на главную страницу.

petitechatte replied:
Неправильный формат для векторного изображения!

По другим изображениям:
1) Абсолютно нет никакого смысла вырезать цветные прямоугольники. Ты только перегружаешь свою страницу и увеличиваешь время загрузки и количество запросов на сервер.
2) То же самое касается цветных кругов. Даже треугольники и крестики можно делать на CSS.
2) Простые малоцветные изображения типа иллюстраций преимуществ или слайдера - это однозначно PNG, возможно даже с прозрачностью.

Надо внимательно пересмотреть форматы для всех картинок!

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      <nav class="main-navigation">
+        <img class="logo-main" src="img/nerds-logo.png" width="160" height="65" alt="Логотип студии Nerds">
+        <ul class="site-navigation">
+          <li><a href="index.html" aria-label="Переход на главную страницу">Студия</a></li>
```
Почему ты решил, что "Студия" - это ссылка на главную? Зачем тогда этот пункт на главной.
В ТЗ ничего не сказано, возможно, подразумевается отдельная страница с подробным описанием студии, так что лучше оставить здесь заглушку.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+        <img class="logo-main" src="img/nerds-logo.png" width="160" height="65" alt="Логотип студии Nerds">
+        <ul class="site-navigation">
+          <li><a href="index.html" aria-label="Переход на главную страницу">Студия</a></li>
+          <li><a href="clients.html" aria-label="Переход на страницу с клиентами нашей фирмы">Клиенты</a></li>
```
Страница не сверстана, клик по ссылке приводит на пустую страницу. Это неудобно, поставь лучше заглушку.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+        <ul class="site-navigation">
+          <li><a href="index.html" aria-label="Переход на главную страницу">Студия</a></li>
+          <li><a href="clients.html" aria-label="Переход на страницу с клиентами нашей фирмы">Клиенты</a></li>
+          <li class="site-navigation-current"><a href="catalog.html" aria-label="Переход на страницу магазина">Магазин</a></li>
```
Ссылка на саму страницу не нужна. Она сбивает пользователей с толку.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+          <li><a href="index.html" aria-label="Переход на главную страницу">Студия</a></li>
+          <li><a href="clients.html" aria-label="Переход на страницу с клиентами нашей фирмы">Клиенты</a></li>
+          <li class="site-navigation-current"><a href="catalog.html" aria-label="Переход на страницу магазина">Магазин</a></li>
+          <li><a href="contacts.html" aria-label="Переход на страницу контактов">Контакты</a></li>
```
Страницы нет, нужно оставить заглушку.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+          <li><a href="contacts.html" aria-label="Переход на страницу контактов">Контакты</a></li>
+        </ul>
+        <ul class="user-navigation">
+          <li class="basket-link" ><a href="#" aria-label="Переход в корзину товаров">Корзина</a></li>
```
В принципе не ошибка, но бросается в глаза лишний пробел в теге.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+
+    <main class="container">
+      <section class="filter">
+        <h2 class="filter-header visualy-hidden">Фильтр товаров.</h2>
```
Непонятно, зачем тебе класс filter-header для невидимого заголовка. 
Кстати, точка в заголовке не ставится.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+    <main class="container">
+      <section class="filter">
+        <h2 class="filter-header visualy-hidden">Фильтр товаров.</h2>
+        <form class="filter-form" action="hhtp://echo.htmlacademy.ru" method="get" aria-label="Форма фильтра товаров">
```
Опять неправильный адрес, как и везде. Ни одна из твоих форм не будет работать.

petitechatte replied:
Форму лучше разбить не на дивы, а на филдсеты, а "стоимость", "сетка", "особенности" станут легендами этих филдсетов.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      <section class="filter">
+        <h2 class="filter-header visualy-hidden">Фильтр товаров.</h2>
+        <form class="filter-form" action="hhtp://echo.htmlacademy.ru" method="get" aria-label="Форма фильтра товаров">
+          <div class="slider-coast-container">
```
Во всем разделе опечатки в названиях классов. Coast - это побережье.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+        <form class="filter-form" action="hhtp://echo.htmlacademy.ru" method="get" aria-label="Форма фильтра товаров">
+          <div class="slider-coast-container">
+            <label for="slider-coast-label">Стоимость:</label>
+            <input type="range" min="1" max="1500" value="50" class="slider-coast" id="slider-coast-label" aria-label="Слайдер переключения стоимости в фильтре">
```
Насколько мне известно, слайдеры с двумя ползунками до сих пор в HTML не появились. Следовательно, ты не можешь использовать обычный input. Его придется собирать вручную (на этапе стилизации) и оживлять с помощью JS (это не входит в ТЗ). Пока что можно оставить как есть. Я добавляла 2 скрытых инпута - для максимальной и минимальной цены. Но можно вообще без инпутов. Тогда "Стоимость" - это не лейбл, а заголовок.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <label for="slider-coast-label">Стоимость:</label>
+            <input type="range" min="1" max="1500" value="50" class="slider-coast" id="slider-coast-label" aria-label="Слайдер переключения стоимости в фильтре">
+            <button class="coast-button-from" aria-label="Кнопка-ползунок переключения минмимальной стоимости в фильтре"></button>
+            <button class="coast-button-before" aria-label="Кнопка-ползунок переключения максимальной стоимости в фильтре"></button>
```
Ползунки ты будешь верстать потом.
Пока что могу сказать, что это вообще не кнопки, они не нажимаются. Для них можно использовать div или span.
Названия классов у тебя совершенно непонятные. Если from еще куда не шло, то before - это "перед", а не "до". (Не зря я тебе говорила, что английский важен, даже если ты верстаешь русскоязычные сайты)
Здесь в классах лучше всего использовать сокращения min и max. Слово button лучше использовать либо в начале, либо в конце названия класса. Да и не кнопка это вообще...

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <input type="range" min="1" max="1500" value="50" class="slider-coast" id="slider-coast-label" aria-label="Слайдер переключения стоимости в фильтре">
+            <button class="coast-button-from" aria-label="Кнопка-ползунок переключения минмимальной стоимости в фильтре"></button>
+            <button class="coast-button-before" aria-label="Кнопка-ползунок переключения максимальной стоимости в фильтре"></button>
+            <label for="coast-from">ОТ</label>
```
В лейблах лучше использовать строчные буквы, а большими их делать с помощью CSS.
Про названия атрибутов смотри выше.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <button class="coast-button-from" aria-label="Кнопка-ползунок переключения минмимальной стоимости в фильтре"></button>
+            <button class="coast-button-before" aria-label="Кнопка-ползунок переключения максимальной стоимости в фильтре"></button>
+            <label for="coast-from">ОТ</label>
+            <output name="min-cost" id="coast-from" aria-label="Значение нижнее цены">0</output>
```
min-cost - вот пример хорошего, грамотного имени. Почему бы его не использовать везде? Можно еще min-price.
В aria-label ты пишешь что-то на языке мастера Йоды. Лучше будет звучать "нижнее значение цены". Или просто "минимальная цена".

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <input type="radio" id="radio-grid-fixed" name="radio-grid-fixed" aria-label="Переключатель выбирает фиксированную сетку">
+            <label for="radio-grid-fixed">Фиксированная</label>
+            <input type="radio" id="radio-grid-rubber" name="radio-grid-rubber" aria-label="Переключатель выбирает резиновую сетку" disabled>
+            <label for="radio-grid-rubber">Резина</label>
```
Несоответствие текста макету.

petitechatte replied:
Важно! Не забывай тестировать, что ты верстаешь: походи по ссылкам, потыкай в форму и т.д.
У тебя можно выбрать одновременно две радиокнопки, а это противоречит их назначению. Радиокнопки только для взаимоисключающего выбора!
Подсказка: неправильные имена инпутов.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+          </div>
+          <div class="filter-checkbox-features">
+            <p>Особенности:</p>
+            <input type="checkbox" id="features-radio-slider" name="features-radio" value="slider" aria-label="Чекбокс выбирает наличие слайдера" checked>
```
Непонятно, зачем везде указано radio. О чем речь вообще?
Форма отправит на сервер пары "name: value". Соответственно, тебе достаточно оставить в названии просто features. "Features: slider" - более чем достаточно. Чекбокс у тебя там или радиобаттон - серверу абсолютно без разницы.
На самом деле, эти данные выдает бэкендер, но тебе нужно представлять хотя бы, как это работает.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <input type="checkbox" id="features-radio-news" name="features-radio" aria-label="Чекбокс выбирает наличие блока новостей" value="news" checked>
+            <label for="features-radio-news">Новости</label>
+            <input type="checkbox" id="features-radio-gallery" name="features-radio" aria-label="Чекбокс выбирает наличие галереи" value="gallery">
+            <label for="features-radio-gallery">Галлерея</label>
```
Опечатка в тексте.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <input type="checkbox" id="features-radio-slider" name="features-radio" value="slider" aria-label="Чекбокс выбирает наличие слайдера" checked>
+            <label for="features-radio-slider">Слайдер</label>
+            <input type="checkbox" id="features-radio-feautures" name="features-radio" aria-label="Чекбокс выбирает наличие спсика преимуществ" value="feautures">
+            <label for="features-radio-feautures">Блок преимуществ</label>
```
features-radio-feautures? Определенно стоит подумать над более подходящим названием.

petitechatte replied:
Кстати, в значении инпута опечатка. Где-то на бэкенде грустит один сервер, который не сможет это обрабатывать.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <input type="checkbox" id="features-radio-gallery" name="features-radio" aria-label="Чекбокс выбирает наличие галереи" value="gallery">
+            <label for="features-radio-gallery">Галлерея</label>
+            <input type="checkbox" id="features-radio-basket" name="features-radio" aria-label="Чекбокс выбирает наличие корзины покупок" value="basket">
+            <label for="features-radio-basket">Галлерея</label>
```
Несоответствие текста макету.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <input type="checkbox" id="features-radio-basket" name="features-radio" aria-label="Чекбокс выбирает наличие корзины покупок" value="basket">
+            <label for="features-radio-basket">Галлерея</label>
+          </div>
+          <button type="submit" name="filter-form-button" value="Показать" aria-label="При нажатии на кнопку отправиться значения формы"></button>
```
Кнопка сверстана неправильно. Она же даже отображается неправильно! У меня такое ощущение, что ты вообще не открываешь свою верстку в браузере...

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <label for="features-radio-basket">Галлерея</label>
+          </div>
+          <button type="submit" name="filter-form-button" value="Показать" aria-label="При нажатии на кнопку отправиться значения формы"></button>
+        </form>
```
Вот что получает сервер, если исправить адрес отправки формы:

* radio-grid-adaptive — on
* radio-grid-fixed — on
* features-radio — [feautures, news]
* filter-form-button — Показать

Как по-твоему, что означает последняя строка и для чего она нужна?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      </section>
+
+      <section class="layouts">
+        <h2 class="header-hidden">Шаблоны веб-сайтов.</h2>
```
В заголовках точка не ставится.

petitechatte replied:
Кстати, в этом разделе у тебя только инструменты для сортировки. Значит, название класса и заголовок должны быть соответствующими.
Сортировку, сами шаблоны, и пагинацию можно обернуть вместе в div или даже в section. Вот тогда текущие названия будут актуальны.

petitechatte replied:
И да, почему header-hidden, а не visually-hidden?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      <section class="layouts">
+        <h2 class="header-hidden">Шаблоны веб-сайтов.</h2>
+        <div class="sort">
+          <p class="sort-caption">Сортировать:</p>
```
Если это caption, то почему тег р?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+        <h2 class="header-hidden">Шаблоны веб-сайтов.</h2>
+        <div class="sort">
+          <p class="sort-caption">Сортировать:</p>
+          <ul class="sort-iteems">
```
Опечатка в названии класса.

petitechatte replied:
Дальше вообще непонятно, что происходит. Почему нет текста в ссылках? Что означает caption? Почему ссылки неактивны? Если действительно какие-то ссылки должны быть неактивными, почему они кликабельны?
Подумай, как должна работать сортировка, и переверстай.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <li class="sort-item-inactive"><a href="#"></a></li>
+            <li class="sort-item-inactive"><a href="#"></a></li>
+          </ul>
+          <ul class="sort-buttoms">
```
Опять опечатка и вообще мало осмысленное название класса.
Какие кнопки? Что они делают?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      </section>
+
+      <section class="gallery">
+        <h2 class="gallery-header visualy-hidden">Галерея готовых шаблонов.</h2>
```
Зачем класс невидимому заголовку? Ты же не будешь его стилизовать.
Точка в заголовке не ставится.

petitechatte replied:
Зачем нужен класс для невидимого заголовка? Ты же не будешь его стилизовать.
Точка в конце заголовка не ставится.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      <section class="gallery">
+        <h2 class="gallery-header visualy-hidden">Галерея готовых шаблонов.</h2>
+        <ul class="gallery-items">
+          <li class="gallery-item"><img src="img/img-sedona.jpg" width="360" height="576" alt="Sedoyf макет."></li>
```
Опечатка в атрибуте alt.

petitechatte replied:
Опечатка в атрибуте alt.
Точки в alt тоже ставить не нужно. А если уж ставишь, то ставь везде, а не как попало.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+
+      <section class="gallery">
+        <h2 class="gallery-header visualy-hidden">Галерея готовых шаблонов.</h2>
+        <ul class="gallery-items">
```
Галерею нужно переверстать так, чтобы ховеры находились в карточках шаблонов! Ведь ховер - это тоже часть карточки, в нем информация о шаблоне. К тому же дизайн может впоследствии измениться, и ховеры станут обычным текстом под картинкой.

Ты должен верстать не то, что нарисовал дизайнер. Разметка отражает логику страницы. А у тебя логики и связи между элементами в данном случае нет никакой.

petitechatte replied:
Галерею нужно переверстать, добавив ховеры в карточки. Ведь это тоже часть карточки, которая при изменении дизайна может отображаться под картинкой.

Разметка должна отображать логику страницы. У тебя сейчас нет ни логики, ни связи между элементами.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+          <li class="gallery-item"><img src="img/img-sedona.jpg" width="360" height="576" alt="Sedoyf макет."></li>
+          <li class="gallery-item"><img src="img/img-pink.jpg" width="360" height="576" alt="Pink шаблон"></li>
+          <li class="gallery-item"><img src="img/img-barbershop.jpg" width="360" height="576" alt="Borodinski макет"></li>
+          <li class="gallery-item"><img src="img/img-mishka.jpg" width="360" height="576" alt="Mishka макет."></li>
```
Точки в alt тоже не нужны. А если уж ставишь, то ставь везде, а не как попало.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+          <li class="gallery-item"><img src="img/img-aplus.jpg" width="360" height="576" alt="Aplus макет."></li>
+          <li class="gallery-item"><img src="img/img-kvast.jpg"  width="360" height="576" alt="Кваст макет."></li>
+        </ul>
+        <div class="pages">
```
Есть устойчивый термин pagination - постраничная навигация в переводе. Лучше использовать его в качестве названия класса.

petitechatte replied:
Есть устойчивый термин pagination - постраничная навигация. Лучше использовать его в качестве класса.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+        </ul>
+        <div class="pages">
+          <ul class="pages-items">
+            <li class="pages-item-current"><button>1</button></li>
```
Почему button? Что делает эта кнопка? Ты в курсе, что у кнопки тип по умолчанию - sumbit, т.е. отправка формы? Какую форму она отправит?
Если ты даешь всем страницам одинаковые классы pages-item, то и здесь его нужно использовать вместе с дополнительным классом для текущей страницы.

petitechatte replied:
Почему кнопка? Что делает эта кнопка? Знаешь ли ты, что у кнопки тип по умолчанию - submit, т.е. отправка формы? Какую форму отправляет эта кнопка?

Дальше насчет классов. Если ты даешь страницам общий класс pages-item, он должен быть у всех страниц. pages-item-current - дополнительный класс.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+            <li class="pages-item-current"><button>1</button></li>
+            <li class="pages-item"><button>2</button></li>
+            <li class="pages-item"><button>3</button></li>
+            <li class="pages-item-next"><button>Следующая</button></li>
```
Если есть кнопка "Следующая", то должна быть и кнопка "Предыдущая", которая не должна отображаться, когда мы находимся на первой странице. Аналогично, когда мы находимся на последней странице, не должно быть кнопки "Следующая". 
Подумай, как это реализовать.

petitechatte replied:
Если есть кнопка "Следующая", то должна быть и кнопка "Предыдущая", которая не отображается, когда мы находимся на первой странице. Аналогично, когда мы попадем на последнюю страницу, не должна отображаться кнопка "Следующая".
Подумай, как это можно реализовать.
Не копируй бездумно элементы с макета.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      </section>
+
+      <div class="sedona-hover">
+        <h3>SEDONA</h3>
```
Всегда думай о том, что может измениться в макете. Сегодня мы хотим название капслоком, а завтра приходит в голову, что луче просто писать с большой буквы. 
Поэтому не используй нигде капслок по умолчанию. Разве что в placeholder и value у инпутов. И то, нужно смотреть, какие там есть сейчас возможности по стилизации, они со временем тоже улучшаются.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+
+      <div class="sedona-hover">
+        <h3>SEDONA</h3>
+        <p>Информацинный сайт для туристов.</p>
```
Опечатка в тексте, лишняя точка.

petitechatte replied:
Опечатка в тексте.
Точка на макете отсутствует, как и в других карточках.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      <div class="a-plus-hover">
+        <h3>A PLUS</h3>
+        <p>Лендинг курсов повышения квалификации.</p>
+        <b class="cost">9900 руб.</b>
```
Числа на макете отображаются с пробелом в качестве разделителя.

petitechatte replied:
Исправь во всех ценах.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+        </div>
+      </section>
+
+      <div class="sedona-hover">
```
Эти дивы можно переносить прямо в карточки. Только класс у всех ховеров должен быть одинаковый. Карточки - однотипные элементы, которые будут генерироваться на PHP.

petitechatte replied:
Все эти дивы нужно разнести по соответствующим карточкам шаблонов, но класс у всех ховеров должен быть одинаковый. Карточка - однотипный элемент, который будет собираться на PHP.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      </div>
+      <div class="a-plus-hover">
+        <h3>A PLUS</h3>
+        <p>Лендинг курсов повышения квалификации.</p>
```
Внимательно смотри на макет. Не ставь точки там, где они не нужны.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      </div>
+      <div class="barbershop-hover">
+        <h3>BARBERSHOP</h3>
+        <p>Сайт салоны красоты. <br> Для мужчин, да.</p>
```
А вот здесь ты сделал все в точности как на макете. Но это тот случай, где лучше исправить опечатку, сделанную в макете.
Тег br тут скорее всего лишний. Лучше поставить потом, когда будешь стилизовать карточки, если текст не будет совпадать с макетом.

petitechatte replied:
Вот здесь ты сделал точно как в макете. А стоило бы исправить опечатку, допущенную дизайнером.
Тег br тут скорее всего не нужен. Лучше добавить его на этапе стилизации карточки, если вдруг текст не будет совпадать с макетом.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      </section>
+
+      <div class="sedona-hover">
+        <h3>SEDONA</h3>
```
Всегда думай о том, как может измениться дизайн. Иногда он меняется прямо в процессе разработки, а значит переверстывать все придется тебе.

Поэтому не делай капслоком то, что можно делать с помощью CSS. Исключение - placeholder и value у инпутов. Но тоже нужно смотреть, возможно и они уже поддаются стилизации.

petitechatte replied:
Смотри ТЗ:
 3.10. Карточка товара: название товара является ссылкой, клик по нему открывает модальное окно с демонстрацией товара (макета нет, модальное окно не делать).

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      <div class="sedona-hover">
+        <h3>SEDONA</h3>
+        <p>Информацинный сайт для туристов.</p>
+        <b class="cost">9900 руб.</b>
```
На макете во всех карточках используется пробел в качестве разделителя в числах.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+    <footer class="main-footer">
+      <div class="footer-contact-map">
+        <div class="footer-contact">
+          <h3>NЁRDS DESIGN STUDIO</h3>
```
Почему именно h3? А где h2? И почему именно это заголовок, а не "Наши контакты", допустим?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+          ул. Б. Конюшенная, д. 19/8
+          </p>
+          <p>тел. +7(812)275-75-75</p>
+          <a class="footer-contact-button" href="#" aria-label="Ссылка на форму отправки данных">Напишите нам</a>
```
"Ссылка на форму отправки данных" - каких данных? Если уж писать пояснения, пиши так, чтобы было понятно, что мы сможем сделать, нажав на ссылку.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+        </div>
+      </div>
+      <div class="footer-social">
+        <b>Давайте дружить это выгодно!</b>
```
Несоответствие макету.
У тебя какое-то слепое пятно в области любых знаков препинания...

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,159 @@
+      </div>
+    </footer>
+  </body>
+</html>
```
Отсутствие пустой строки в конце файла.

petitechatte commented on img/background-social.svg:
```diff
@@ -0,0 +1,16 @@
+    </filter>
+  </defs>
+  <circle id="Shape_2_copy_3" data-name="Shape 2 copy 3" class="cls-1" cx="39.125" cy="39.125" r="39.125"/>
+</svg>
```
Возможно, я пока чего-то не понимаю, но весь этот код можно заменить двумя строчками на CSS.

petitechatte commented on img/slider-button-no-active.svg:
```diff
@@ -0,0 +1,10 @@
+    </style>
+  </defs>
+  <circle id="Shape_1_copy_2" data-name="Shape 1 copy 2" class="cls-1" cx="9" cy="8.984" r="9"/>
+</svg>
```
Тоже не вижу особого смысла в этой картинке, но, возможно, в этом случае использование картинки более удобно. Но обычно это делается псевдоэлементом.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
   <body>
+    <header class="main-header">
+      <nav class="main-navigation">
+        <img class="logo-main" width="160" height="65" src="img/nerds-logo.png" alt="Логотип студии Nerds">
```
Неправильный формат для векторного изображения!

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+    <header class="main-header">
+      <nav class="main-navigation">
+        <img class="logo-main" width="160" height="65" src="img/nerds-logo.png" alt="Логотип студии Nerds">
+        <ul class="site-navigation">
```
Замечания по навигации я уже писала для внутренней страницы. Всем страницам, кроме магазина, нужно проставить заглушки.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        </ul>
+      </nav>
+
+      <section class="slide-items">
```
Неподходящее название класса для раздела. Обычно так именуют списки.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+      </nav>
+
+      <section class="slide-items">
+        <h2 class="slider-heading-hidden">Слайдер рекламный обыкновенный</h2>
```
Неподходящее название. Подумай, что именно сообщается на слайдах, так и назови.

petitechatte replied:
И что с классом заголовка? Здесь должен быть классический visually-hidden.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+
+      <section class="slide-items">
+        <h2 class="slider-heading-hidden">Слайдер рекламный обыкновенный</h2>
+        <div class="slide-item-first">
```
У всех однотипных элементов должен быть одинаковый класс, который будет описывать их одинаковые характеристики, например, slider-item. Затем можно добавлять дополнительные классы для отдельных слайдов. Или даже id, если делать слайдер на радиокнопках.

petitechatte replied:
Кстати, последовательность однотипных элементов - это список! То есть каждый слайд - это элемент списка.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+      <section class="slide-items">
+        <h2 class="slider-heading-hidden">Слайдер рекламный обыкновенный</h2>
+        <div class="slide-item-first">
+          <b class="slide-lead">Долго, дорого скурпулезно.</b>
```
Опечатка в тексте.
Я, кстати, убирала здесь запятую, т.к. не согласна с дизайнером.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        <h2 class="slider-heading-hidden">Слайдер рекламный обыкновенный</h2>
+        <div class="slide-item-first">
+          <b class="slide-lead">Долго, дорого скурпулезно.</b>
+          <p class="slide-text">Математически выверенный дизайн длдя вашего сайта или мобильного приложения.</p>
```
Опечатка в тексте!
Откуда они берутся? Почему ты не копируешь текст из макета?

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        <div class="slide-item-first">
+          <b class="slide-lead">Долго, дорого скурпулезно.</b>
+          <p class="slide-text">Математически выверенный дизайн длдя вашего сайта или мобильного приложения.</p>
+          <a class="slide-link" aria-label="Ссылка на переход на рекламную информацию">Узнать больше</a>
```
"Ссылка на переход" - тавтология. Мы всегда по ссылкам куда-то переходим.

petitechatte replied:
Всем ссылкам нужны href с заглушкой!

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          <b class="slide-lead">Долго, дорого скурпулезно.</b>
+          <p class="slide-text">Математически выверенный дизайн длдя вашего сайта или мобильного приложения.</p>
+          <a class="slide-link" aria-label="Ссылка на переход на рекламную информацию">Узнать больше</a>
+          <ul class="slide-controls">
```
Слайдер вынеси из слайдов. Это отдельная панель управления слайдами.
Внутри элементов списка можно делать обычные кнопки или радиобаттоны.  Им желательны какие-то подписи, можно просто 1, 2, 3.

petitechatte replied:
Ссылку кстати тоже можно вынести, если она ведет на одну и ту же страницу. Но в ТЗ об этом нигде не говорится. Можно оставить ссылку на каждом слайде.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+    </header>
+
+    <main class="container">
+      <h1 class="conrainer-heading visualy-hidden">Веб-студия Nerds</h1>
```
Невидимому заголовку не нужны другие классы.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+    <main class="container">
+      <h1 class="conrainer-heading visualy-hidden">Веб-студия Nerds</h1>
+      <section class="advantages">
+        <h2 class="advantages-heading visualy-hidden">Преимущества студии Nerds</h2>
```
Невидимому заголовку не нужны другие классы.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        <ul class="advantages-items">
+          
+          <li class="advantages-item">
+            <a class="advantages-item-image1" href="#"><img src="img/advantages-illustration-1.jpg" width="300" height="146" alt="Веб-сайты студии"></a>
```
В ТЗ не говорится, что картинка является ссылкой. Ссылкой является "кнопка" заказать. Картинка вообще декоративная, она вставляется фоном. И это грубая ошибка, кстати.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          <li class="advantages-item">
+            <a class="advantages-item-image1" href="#"><img src="img/advantages-illustration-1.jpg" width="300" height="146" alt="Веб-сайты студии"></a>
+            <h3>Веб-сайты</h3>
+            <b class="triumph">Мир никогда не будет прежним после того как увидит ваш сайт!</b>
```
Здесь можно оставить просто p. И еще хорошо бы подправить за дизайнером пунктуацию.
Класс непонятный, можно вообще оставить текст без класса.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        <h2 class="advantages-heading visualy-hidden">Преимущества студии Nerds</h2>
+        <ul class="advantages-items">
+          
+          <li class="advantages-item">
```
Блоки преимуществ отличаются оформлением, поэтому, кроме общего класса advantages-item, каждому нужно добавить уникальный класс.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+            <a class="advantages-item-image1" href="#"><img src="img/advantages-illustration-1.jpg" width="300" height="146" alt="Веб-сайты студии"></a>
+            <h3>Веб-сайты</h3>
+            <b class="triumph">Мир никогда не будет прежним после того как увидит ваш сайт!</b>
+            <a class="advantages-item-button1" href="#">Заказать</a>
```
С кнопками то же самое: должен быть общий класс для всех кнопок. Возможно даже универсальный класс button для всех кнопок и ссылок на сайте. Затем, если нужно, класс для кнопок преимуществ. И наконец, уникальный класс для каждой кнопки, но его можно даже не ставить, а использовать класс для преимущества.

Это все будет понятно, когда начнешь стилизовать.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          </li>
+          <li class="advantages-item">
+            <a class="advantages-item-image3" href="#"><img src="img/advantages-illustration-3.jpg" width="300" height="146" alt="Презентации студии."></a>
+            <h3>Веб-сайты</h3>
```
Несоответствие макету!
Не забывай все время смотреть в браузере, что ты наверстал!

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+      </section>
+
+      <section class="studio">
+        <h2 class="studio-handing visualy-hidden">Инормация о студии.</h2>
```
Опечатка в тексте.
Точка в заголовке не ставится.
Дополнительный класс при наличии visualy-hidden не нужен. Хотя, может, я тут и не права... Кстати, что такое handing?

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        <b class="studio-lead">Мы — маленькая, но гордая дизайн-студия из Краснодара.</b>
+        <p>Исповедуем принципы минимализма и чистоты. Ставим математический расчет
+        превыше креатива. Работаем не покладая рук, как роботы.</p>
+        <ul class="services-items">
```
В классах для списков обычно используют слово list (список), например, services-list). Это помогает не путаться между очень похожими классами services-items и services-item.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          </li>
+          <li class="advantages-item">
+            <a class="advantages-item-image2" href="#"><img src="img/advantages-illustration-2.jpg" width="300" height="146" alt="Приложения студии."></a>
+            <h3>Веб-сайты</h3>
```
Несоответствие макету!

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        <ul class="services-items">
+          <li class="services-item">Веб-сайтов любой сложности</li>
+          <li class="services-item">Мобильных приложений для IOS и Android</li>
+          <li class="services-item">Слайдшоу и видео для корпоративных презентаций</li>
```
Несоответствие макету!
Список не имеет смысла, поскольку отсутствует часть текста.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          <li class="services-item">Мобильных приложений для IOS и Android</li>
+          <li class="services-item">Слайдшоу и видео для корпоративных презентаций</li>
+        </ul>
+        <img src="img/nerds-illustration.jpg" width="360" height="208" alt="Логотип веб-студии Nerds">
```
Декоративная картинка. Грубая ошибка, нужно удалить. Кстати, формат изображения тоже выбран неправильно.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          <li class="services-item">Слайдшоу и видео для корпоративных презентаций</li>
+        </ul>
+        <img src="img/nerds-illustration.jpg" width="360" height="208" alt="Логотип веб-студии Nerds">
+        <span>С 2004 года любим точность во всем:</span>
```
Возможно, здесь лучше использовать тег р.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        </ul>
+        <img src="img/nerds-illustration.jpg" width="360" height="208" alt="Логотип веб-студии Nerds">
+        <span>С 2004 года любим точность во всем:</span>
+        <table class="studio-statistic">
```
Опечатка в названии класса.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          <tr>
+            <td>146%</td>
+            <td>Уровень самоотдачи</td>
+          </tr>
```
По хорошему, "Уровень самоотдачи", это заголовок таблицы, он должен идти первым и обозначаться тегом `th`. Но так тебе будет тяжело сверстать, поэтому можешь оставить таблицу как есть.

**Грубая ошибка!** `tr` - table row, строка таблицы. А ты в ней описываешь столбец! Опять-таки, посмотри, как выглядит то, что ты сверстал!

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          </tr>
+        </table>
+        <section class="partners">
+          <h2 class="partners-header visualy-hidden">Наши партнеры</h2>
```
Лишний класс у невидимого заголовка.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+        <section class="partners">
+          <h2 class="partners-header visualy-hidden">Наши партнеры</h2>
+          <ul class="partners-list">
+            <li class="partners-list-item"><a href="#" title="HTMLAcademy" aria-label="Переход на страницу Академии"><img src="img/logo-1.png" width="199" height="68" alt="Логотип HTMLAcademy"></a></li>
```
Смотри ТЗ:
3.6. Логотип Академии в блоке с партнёрами ведёт на лендинг интенсива «Профессиональный HTML и CSS, уровень 1».

Хорошо бы еще и называния файлов с логотипами делать более осмысленными.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+          <li><a class="social-button" href="https://www.instagram.com/" aria-label="Переход на нашу страницу в Instagram">Инстаграм</a></li>
+        </ul>
+      </div>
+    </footer>
```
Замечания по футеру описаны на странице каталога.

petitechatte replied:
Отсутствует разметка модального окна!

petitechatte commented on index.html:
```diff
@@ -1,12 +1,150 @@
+    </footer>
   </body>
-</html>
+</html>
```
Отсутствует пустая строка в конце файла.

petitechatte commented on workspace.code-workspace-nerds.code-workspace:
```diff
@@ -0,0 +1,10 @@
+	"settings": {
+		"liveServer.settings.multiRootWorkspaceName": "css"
+	}
+}
```
Хоть это и служебный файл, GitHub требует пустую строку в конце.
Кстати, есть подозрение, что этот файл тоже нужно добавить в .gitignore.

petitechatte reviewed (changes requested):
Очень, очень грязная верстка.
Я потратила кучу времени из-за того, что ты даже не открыл свои файлы в браузере!
В следующий раз я не буду так подробно расписывать ошибки, если ты будешь сдавать работу в таком виде.

Большие недоработки по теме "Изображения". Пересмотри еще раз теорию, особенно то, что касается выбора формата изображения, а также разницы между контентным и декоративным изображением.

Nomad48JMS committed:
Подправил некоторые ошибки. Переписал классы index по БЭМ.

Подправил некоторые ошибки. Переписал классы index по БЭМ.

Nomad48JMS committed:
Revert "Подправил некоторые ошибки. Переписал классы index по БЭМ."

This reverts commit 4cd4cd99e0cfbdda40f4141a14dc8330d0c03f11.

Nomad48JMS committed:
Revert "Revert "Подправил некоторые ошибки. Переписал классы index по БЭМ.""

This reverts commit 87c9c6805ff2fcf0777728b5543ee6c1f6d3091c.

Nomad48JMS committed:
Исправил ошибки и замечания. Сделал проверку и валидацию. Попробовал сделать классы по БЭМ. Убрал лишние файлы и переделано много.

Исправил ошибки и замечания. Сделал проверку и валидацию. Попробовал сделать классы по БЭМ. Убрал лишние файлы и переделано много.

Nomad48JMS committed:
Убрал лишние aria-label из навигации. Думаю что они здесь лишние.

Убрал лишние aria-label из навигации. Думаю что они здесь лишние.

Nomad48JMS committed:
Пару отступов убрал

Пару отступов убрал

Nomad48JMS committed:
Changed markup, classes. Changed some of the images. Made a piece of basic styling.

Изменил разметку, классы. Изменил часть изображений. Сделал кусок базовой стилизации.

Nomad48JMS committed:
Made on assignment 5 chapter 1 course. HTML Run on the validator everything works. Fixed bugs and BEM classes. Made basic page styling. Checked it out. I ran the CSS for validation - everything works.

Сделал по заданию 5 главы 1 курса. HTML Прогнал на валидаторе все работает.
Исправил ошибки и классы по БЭМ Сделал базовую стилизацию страницы. Проверил. Прогнал CSS на валидацию -все работает.

Nomad48JMS committed:
Сделал сетки на флексах index

Сделал сетки на флексах index

Nomad48JMS committed:
ver

petitechatte commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+#������������ ����� �� ����� layout
```
Непонятная лишняя строка

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+
+  <body>
+    <header class="main-header">
+      <nav class="main-header__main-navigation main-navigation container-centerer">
```
Сомнительное слово "centerer", не уверена, что оно вообще существует.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+        <ul class="main-navigation__site-navigation">
+          <li><a href="#">Студия</a></li>
+          <li><a href="#">Клиенты</a></li>
+          <li><a class="main-navigation__link--current">Магазин</a></li>
```
С точки зрения БЭМ выглдит странно. На данной ссылке есть класс с модификатором, а на остальных ссылках общего класса нет. Есть он (main-navigation__link) только на логотипе, который сильно отличается от остальной навигации.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          <li><a href="#">Контакты</a></li>
+        </ul>
+        <ul class="main-navigation__user-navigation">
+          <li class="main-navigation__basket-link"><a href="#" aria-label="Переход в корзину товаров">Корзина</a></li>
```
Типичная ошибка локализации: по-русски мы говорим "корзина", а у англоговорящих - это **cart**. Отсюда и иконка тележки на сайтах.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+      <h1>Магазин готовых шаблонов</h1>
+    </header>
+
+    <main class="container-catalog container">
```
В чем смысл класса container-catalog?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+      <section class="container-catalog__filter filter">
+        <h2 class="visually-hidden">Фильтр товаров</h2>
+        <form class="filter__form form" action="http://echo.htmlacademy.ru" method="get" aria-label="Форма фильтра товаров">
+          <fieldset class="form__slider">
```
Я бы основывала название данного блока не на слайдере, а на том, что он про цены.

petitechatte replied:
```suggestion
          <fieldset class="form__cost cost">
```

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <label class="form__label" for="coast-from">от</label>
+            <output class="form__min-cost" name="form-min-cost" id="coast-from" aria-label="Минимальная цена">0</output>
+            <label class="form__label" for="cost-up-to">до</label>
+            <output class="form__max-cost" name="form-max-cost" id="cost-up-to" aria-label="Максимальная цена">15000</output>
```
```suggestion
            <legend class="form__legend" >Стоимость:</legend>
            <div class="cost__slider">
                <div class="cost__slider-bar">
                <div class="cost__slider-progress">
                <span class="cost__slider-control" id="min-cost-control"></span>
                <span class="cost__slider-control" id="max-cost-control"></span>
            </div>
            <label class="cost__label" for="min-cost-value">от</label>
            <output class="cost__min-cost" name="min-cost" id="min-cost-value" aria-label="Минимальная цена">0</output>
            <label class="cost__label" for="max-cost-value">до</label>
            <output class="cost__max-cost" name="max-cost" id="max-cost-value" aria-label="Максимальная цена">15 000</output>
```

petitechatte replied:
15 000 - на макете с пробелом

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <label class="form__label" for="radio-block-fixed">Фиксированная</label>
+            <input class="form__radio" type="radio" id="radio-block-rubber" name="radio-block" value="rubber" aria-label="Переключатель выбирает резиновую сетку" disabled>
+            <label class="form__label" for="radio-block-rubber">Резина</label>
+          </fieldset>
```
```suggestion
          <fieldset class="form__grid">
            <legend class="form__legend">Сетка:</legend>
            <input class="form__radio" type="radio" id="grid-adaptive" name="grid" value="adaptive" aria-label="Переключатель выбирает адаптивную сетку" checked>
            <label class="form__label" for="grid-adaptive">Адаптивная</label>
            <input class="form__radio" type="radio" id="grid-fixed" name="grid" value="fixed" aria-label="Переключатель выбирает фиксированную сетку">
            <label class="form__label" for="grid-fixed">Фиксированная</label>
            <input class="form__radio" type="radio" id="grid-rubber" name="grid" value="rubber" aria-label="Переключатель выбирает резиновую сетку" disabled>
            <label class="form__label" for="grid-rubber">Резина</label>
          </fieldset>
```

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <label class="form__label" for="cost-up-to">до</label>
+            <output class="form__max-cost" name="form-max-cost" id="cost-up-to" aria-label="Максимальная цена">15000</output>
+          </fieldset>
+          <fieldset class="form__radio-block">
```
Название блока должно опираться на его смысл, а не набор тегов. Если в форму добавить еще один блок переключателей, то как их различать? То же самое с чекбоксами ниже.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          </fieldset>
+          <fieldset class="form__radio-block">
+            <legend class="form__radio-llegend">Сетка:</legend>
+            <input class="form__radio" type="radio" id="radio-block-adaptive" name="radio-block" value="adaptive" aria-label="Переключатель выбирает адаптивную сетку" checked>
```
name - это то, что идет на сервер, т.е. интегрировано в бэкенд. Не нужно пытаться подогнать name под БЭМ. Наоборот, пара name/value должна иметь смысл в отрыве от твоей верстки.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <label class="form__label" for="checkbox-gallery">Галерея</label>
+            <input class="form__checkbox" type="checkbox" id="checkbox-basket" name="basket-checkbox" value="basket" aria-label="Чекбокс выбирает наличие корзины покупок">
+            <label class="form__label" for="checkbox-basket">Корзина</label>
+          </fieldset>
```
```suggestion
          <fieldset class="form__features">
            <legend class="form__legend">Особенности:</legend>
            <input class="form__checkbox" type="checkbox" id="checkbox-slider" name="slider" checked aria-label="Чекбокс выбирает наличие слайдера">
            <label class="form__label" for="checkbox-slider">Слайдер</label>
            <input class="form__checkbox" type="checkbox" id="checkbox-features" name="features" aria-label="Чекбокс выбирает наличие списка преимуществ">
            <label class="form__label" for="checkbox-features">Блок преимуществ</label>
            <input class="form__checkbox" type="checkbox" id="checkbox-news" name="news" checked aria-label="Чекбокс выбирает наличие блока новостей">
            <label class="form__label" for="checkbox-news">Новости</label>
            <input class="form__checkbox" type="checkbox" id="checkbox-gallery" name="gallery" aria-label="Чекбокс выбирает наличие галереи" >
            <label class="form__label" for="checkbox-gallery">Галерея</label>
            <input class="form__checkbox" type="checkbox" id="checkbox-cart" name="cart" aria-label="Чекбокс выбирает наличие корзины покупок">
            <label class="form__label" for="checkbox-cart">Корзина</label>
          </fieldset>
```

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          </fieldset>
+          <fieldset class="form__checkbox-block">
+            <legend class="">Особенности:</legend>
+            <input class="form__checkbox" type="checkbox" id="checkbox-slider" name="slider-checkbox" value="slider" checked aria-label="Чекбокс выбирает наличие слайдера">
```
Name должен быть содержательным, а value для чекбоксов вообще не обязательны, т.к. автоматически подставляется on/off.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <label class="form__label" for="checkbox-features">Блок преимуществ</label>
+            <input class="form__checkbox" type="checkbox" id="checkbox-news" name="news-checkbox" value="news" checked aria-label="Чекбокс выбирает наличие блока новостей">
+            <label class="form__label" for="checkbox-news">Новости</label>
+            <input class="form__checkbox" type="checkbox" id="checkbox-gallery" name="gallery-checkbox" value="gallery" aria-label="Чекбокс выбирает наличие галлереи" >
```
Опечатка в aria-label

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <label class="form__label" for="checkbox-news">Новости</label>
+            <input class="form__checkbox" type="checkbox" id="checkbox-gallery" name="gallery-checkbox" value="gallery" aria-label="Чекбокс выбирает наличие галлереи" >
+            <label class="form__label" for="checkbox-gallery">Галерея</label>
+            <input class="form__checkbox" type="checkbox" id="checkbox-basket" name="basket-checkbox" value="basket" aria-label="Чекбокс выбирает наличие корзины покупок">
```
Корзина - cart

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <input class="form__checkbox" type="checkbox" id="checkbox-basket" name="basket-checkbox" value="basket" aria-label="Чекбокс выбирает наличие корзины покупок">
+            <label class="form__label" for="checkbox-basket">Корзина</label>
+          </fieldset>
+          <button class="form__form-button" type="submit" name="filter-form-button" aria-label="При нажатии на кнопку отправиться значения формы">Показать</button>
```
Я бы добавила сюда еще глобальный класс button.
Name кнопке не нужен, зачем передавать его на сервер?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+      <section class="container-catalog__layouts">
+        <h2 class="visually-hidden">Шаблоны веб-сайтов</h2>
+        <div class="container-catalog__layouts-sort layouts-sort">
+          <p class="layouts-sort__title">Сортировать:</p>
```
На макете есть несколько одинаковых по стилям заголовков: легенды в форме, сортировка, названия шаблонов. Возможно будут добавляться еще такие заголовки для однообразия дизайна. Можно дать им всем один класс.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <li class="layouts-sort__link"><a href="#">По типу</a></li>
+            <li class="layouts-sort__link"><a href="#">По назначению</a></li>
+          </ul>
+          <ul class="layouts-sort__items">
```
Непонятное название класса. Что имеется в виду под items?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          <ul class="layouts-sort__items">
+            <li class="layouts-sort__item--current"><button class="layouts-sort__button-down" type="button"></button></li>
+            <li class="layouts-sort__item"><button class="layouts-sort__button-up" type="button"></button></li>
+          </ul>
```
```suggestion
          <ul class="layouts-sort__type">
            <li class="layouts-sort__link layouts-sort__link--current"><a>По цене</a></li>
            <li class="layouts-sort__link"><a href="#">По типу</a></li>
            <li class="layouts-sort__link"><a href="#">По названию</a></li>
          </ul>
          <ul class="layouts-sort__direction">
            <li class="layouts-sort__item layouts-sort__item--current"><a class="layouts-sort__direction-link layouts-sort__direction-link--descending"></a></li>
            <li class="layouts-sort__item"><a href="#" class="layouts-sort__direction-link layouts-sort__direction-link--ascending"></a></li>
          </ul>
```

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          <ul class="layouts-sort__links">
+            <li class="layouts-sort__link--current"><a href="#">По цене</a></li>
+            <li class="layouts-sort__link"><a href="#">По типу</a></li>
+            <li class="layouts-sort__link"><a href="#">По назначению</a></li>
```
Несоответствие текста макету!

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+        <div class="container-catalog__layouts-sort layouts-sort">
+          <p class="layouts-sort__title">Сортировать:</p>
+          <ul class="layouts-sort__links">
+            <li class="layouts-sort__link--current"><a href="#">По цене</a></li>
```
В ТЗ, к сожалению, не прописано поведение сортировки. Приходится догадываться, и можно рассмотреть разные варианты:
1. Треугольники - это контролы для изменения направления сортировки (по возрастанию или убыванию). Ты выбрал этот вариант. В таком случае текущая ссылка должна быть неактивна.
2. Треугольники - это просто индикаторы текущей сортировки. В этом случае изменение направления сортировки происходит по повторному клику на ссылку. Тогда все ссылки, включая текущую, должны быть активны.

Не забывай добавлять 2 класса: обычный и с модификатором.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          </ul>
+          <ul class="layouts-sort__items">
+            <li class="layouts-sort__item--current"><button class="layouts-sort__button-down" type="button"></button></li>
+            <li class="layouts-sort__item"><button class="layouts-sort__button-up" type="button"></button></li>
```
Оба блока контролов в сортировке ведут себя одинаково - изменяют порядок карточек на странице. Поэтому они все должны быть либо ссылками, либо кнопками. Если мы планируем делать фолбэк на случай отвалившегося скрипта, то лучше все делать ссылками.

Опять же, не забывай добавлять 2 класса: обычный и с модификатором.

petitechatte replied:
А еще здесь все чудовищно плохо с доступностью. Сортировка может работать даже с отвалившимися стилями, поэтому ссылкам нужен текст!

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+              </div>
+            </li>
+            <li class="gallery__item">
+              <img src="img/img-pink.jpg" width="360" height="576" alt="Pink шаблон">
```
А почему везде макет, а тут шаблон?

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <li class="gallery__item">
+              <img src="img/img-sedona.jpg" width="360" height="576" alt="Sedona макет">
+              <div class="gallery__hover">
+                <a class="gallery__title-link" href="#">Sedona</a>
```
Это заголовок, нужно обозначить соответствующим тегом.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+              <div class="gallery__hover">
+                <a class="gallery__title-link" href="#">Sedona</a>
+                <p>Информацинный сайт для туристов</p>
+                <b class="gallery__cost">9 900 руб</b>
```
После сокращения ставится точка (смотри макет).

petitechatte replied:
Судя по стайлгайду, цена - это ссылка!

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <li class="gallery__item">
+              <img src="img/img-barbershop.jpg" width="360" height="576" alt="Barbershop Borodinski макет">
+              <div class="gallery__hover">
+                <a class="gallery__title-link" href="#">Barbershop Borodinski</a>
```
Несоответствие текста макету!

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+              <img src="img/img-barbershop.jpg" width="360" height="576" alt="Barbershop Borodinski макет">
+              <div class="gallery__hover">
+                <a class="gallery__title-link" href="#">Barbershop Borodinski</a>
+                <p>Сайт салон красоты <span>Для мужчин, да</span></p>
```
Почему текст обернут в `span`? Второе предложение ничем не отличается от первого. К тому же, `span` и `div` без классов не имеют смысла.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+              <li class="pagination__item"><button class="pagination__button" type="button" value="2">2</button></li>
+              <li class="pagination__item"><button class="pagination__button" type="button" value="3">3</button></li>
+              <li class="pagination__item-next"><button class="pagination__button" type="button" value="next">Следующая</button></li>
+            </ul>
```
Общий класс **pagination__item** должен быть у всех ссылок.
pagination__item-previous - это модификация, значит, должно быть **pagination__item--previous**. И вместо класса **visually-hidden** я бы добавила просто атрибут **hidden**.

Кнопками пагинация не может быть никак, т.к. подгружает другую страницу или новые данные. Это ссылки.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+
+    <footer class="main-footer container-centerer">
+      <div class="main-footer__contacts contacts ">
+        <div class="wrapper-map">
```
Этот класс и не глобальный, и не по БЭМ.

petitechatte replied:
И что он делает, тоже не очень понятно.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <span class="contacts__marker"></span>
+          </div>
+        </div>
+        <div class="contacts__contacts">
```
contacts__contacts - плохо звучит. Можно заменить на contacst__address.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          </div>
+        </div>
+        <div class="contacts__contacts">
+          <h3>Nerds design studio</h3>
```
Внимательно посмотри, как они пишут название.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+        </div>
+        <div class="contacts__contacts">
+          <h3>Nerds design studio</h3>
+          <p>191186, Санкт-Петербург,<br>
```
`br` здесь не обязателен, но, в принципе, можно оставить.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          ул. Б. Конюшенная, д. 19/8
+          </p>
+          <p>тел. +7(812)275-75-75</p>
+          <a class="contacts__button" href="#" aria-label="Ссылка на форму отправки данных">Напишите нам</a>
```
Стоит добавить глобальный класс button.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+
+        <div class="main-footer__social social">
+          <b>Давайте дружить это выгодно!</b>
+          <p>Скидка 10% для друзей из социальных сетей</p>
```
На макете точка в конце предложения.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+          <ul class="social__social-links">
+            <li><a class="social__link" href="https://vk.com/" aria-label="Переход на нашу страницу в Вконтакте"><img src="../img/vk-icon.png" alt="Иконка ВК">Вконтакте</a></li>
+            <li><a class="social__link" href="https://www.facebook.com/" aria-label="Переход на нашу страницу в Facebook"><img src="../img/fb-icon.png" alt="Иконка FB">Facebook</a></li>
+            <li><a class="social__link" href="https://www.instagram.com/" aria-label="Переход на нашу страницу в Instagram"><img src="../img/insta-icon.png" alt="Иконка Insta">Instagram</a></li>
```
Иконки не подгрузились. Кстати, их нужно добавлять фоном, а не в разметку.

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+            <li><a class="social__link" href="https://www.instagram.com/" aria-label="Переход на нашу страницу в Instagram"><img src="../img/insta-icon.png" alt="Иконка Insta">Instagram</a></li>
+          </ul>
+        </div>
+      </footer>
```
Отсутствует разметка модального окна!

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+          <li><a href="#">Контакты</a></li>
+        </ul>
+        <ul class="main-navigation__user-navigation">
+          <li class="main-navigation__basket-link"><a href="#" aria-label="Переход в корзину товаров">Корзина</a></li>
```
Корзина - cart

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+      </nav>
+
+      <section class="main-header__slide-items slide-items container-centerer">
+        <h2 class="visually-hidden">Слайдер</h2>
```
Не очень удачное название, т.к. описывает механизм показа информации, но не ее суть. Сюда подойдет название типа "Наши принципы", что-то в этом роде.

petitechatte replied:
Слайды, кстати, хорошо бы сделать списком.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+
+      <section class="main-header__slide-items slide-items container-centerer">
+        <h2 class="visually-hidden">Слайдер</h2>
+        <div class="slide-items__slide slide slide-1">
```
Если ты хочешь все по БЭМ, то тогда **slide--1**.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+              <ul class="slide__slide-controls slide-controls">
+                <li class="slide-controls__switcher"><button class="slide-controls__button-1 slide-controls__button" type="button" aria-label="Кнопка переключателя"></button></li>
+                <li class="slide-controls__switcher"><button class="slide-controls__button-2 slide-controls__button" type="button" aria-label="Кнопка переключателя"></button></li>
+                <li class="slide-controls__switcher"><button class="slide-controls__button-3 slide-controls__button" type="button" aria-label="Кнопка переключателя"></button></li>
```
Здесь тоже не забвывай про БЭМ:
slide-controls__button--1
slide-controls__button--2
slide-controls__button--3

petitechatte replied:
Еще здесь можно добавить доступности.
В aria-label сейчас абсолютно одинаковый текст, непонятно, что делают эти кнопки. Да и внутри ссылок можно прописать "Слайд 1, 2, 3", завернув в **span.visually-hidden**.

petitechatte replied:
Тут кстати, большой вопрос, нужны ли aria-label, если ты прячешь слайды с помощью visually-hidden (то есть их все видит "читалка" сразу). Вот если использовать атрибут hidden, тогда да, нужно озвучивать, что это за кнопки.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+      <section class="container__advantages advantages">
+        <h2 class="visually-hidden">Преимущества студии Nerds</h2>
+        <ul class="advantages__items">
+          <li class="advantages__item item-1">
```
Классы по БЭМ: "advantages__item advantages__item--1"

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+          <li class="advantages__item item-1">
+            <h3>Веб-сайты</h3>
+            <b class="advantages__lead">Мир никогда не будет прежним после того как увидит ваш сайт!</b>
+            <a class="advantages__button1 advantages-button" href="#">Заказать</a>
```
Опять-таки, по БЭМ - advantages__button--1.

Но я бы сделала по-другому:
Кнопки на сайте все похожие, поэтому основные характеристики можно задать глобальным классом **button**.
Кнопки в разных блоках различаются только цветами, поэтому достаточно использовать каскад:
.item--1 .button (и т.п.)
То есть не давать БЭМ-классов кнопкам вообще.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+          <h3 class="studio__title-items">Выполняем заказы на разработку:</h3>
+          <ul class="studio__items">
+            <li class="studio__services-item">Веб-сайтов любой сложности</li>
+            <li class="studio__services-item">Мобильных приложений для IOS и Android</li>
```
iOs

petitechatte commented on catalog.html:
```diff
@@ -0,0 +1,188 @@
+              <img src="img/img-pink.jpg" width="360" height="576" alt="Pink шаблон">
+              <div class="gallery__hover">
+                <a class="gallery__title-link" href="#">Pink app</a>
+                <p>Продуктовый лендинг приложения для IOS и Android</p>
```
iOs

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+          </ul>
+        </div>
+        <div class="studio__information information">
+          <span>С 2004 года любим точность во всем:</span>
```
`Span` без класса бесполезен.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+          <li class="partners__list-item"><a href="#" title="HTMLAcademy" aria-label="Переход на страницу Академии"><img src="img/logo-1.png" width="199" height="68" alt="Логотип HTMLAcademy"></a></li>
+          <li class="partners__list-item"><a href="#" title="Barbershop Borodinski" aria-label="Переход на страницу Барбершопа"><img src="img/logo-2.png" width="199" height="68" alt="Логотип Barbershop Borodinski"></a></li>
+          <li class="partners__list-item"><a href="#" title="Pink" aria-label="Переход на страницу Pink"><img src="img/logo-3.png" width="199" height="68" alt="Логотип Pink"></a></li>
+          <li class="partners__list-item"><a href="#" title="Mishka" aria-label="Переход на страницу Mishka"><img src="img/logo-4.png" width="199" height="68" alt="Логотип Mishka"></a></li>
```
Грубая ошибка. Логотипы - контентные изображения, обязательно вставляются в верстку.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
 
-    <p>Репозиторий создан для обучения верстке Саньки Петрова.</p>
+    <footer class="main-footer container-centerer">
+      <div class="main-footer__contacts contacts ">
```
Лишний пробел

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+            <span class="contacts__marker"></span>
+          </div>
+        </div>
+        <div class="contacts__contacts">
```
Неудачное название класса.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+          ул. Б. Конюшенная, д. 19/8
+          </p>
+          <p>тел. +7(812)275-75-75</p>
+          <a class="contacts__button" href="#" aria-label="Ссылка на форму отправки данных">Напишите нам</a>
```
Имеет смысл сделать глобальный класс button.

petitechatte commented on index.html:
```diff
@@ -1,12 +1,177 @@
+        <ul class="social__social-links">
+          <li><a class="social__link" href="https://vk.com/" aria-label="Переход на нашу страницу в ВК"><img src="../img/vk-icon.png" alt="Иконка ВК">Вконтакте</a></li>
+          <li><a class="social__link" href="https://www.facebook.com/" aria-label="Переход на нашу страницу в Facebook"><img src="../img/fb-icon.png" alt="Иконка FB">Facebook</a></li>
+          <li><a class="social__link" href="https://www.instagram.com/" aria-label="Переход на нашу страницу в Instagram"><img src="../img/insta-icon.png" alt="Иконка Insta">Instagram</a></li>
```
Картинки нужно добавить фоном и проверить пути к файлам.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+/* ?Проверил. Прогнал CSS на валидацию -все работает. */
+
+/* !!! Обнуление */
```
Обнуление в этом проекте использовать не надо.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	padding: 0;
+	overflow: hidden;
+	border: 0;
+	clip: rect(0 0 0 0)
```
Пропущена точка с запятой после правила.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	font-weight: 400;
+
+	background-color: #ffffff;
+	background-image: url(../img/background_index.png);
```
Грубая ошибка. Фон просто белого цвета.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+a {
+	text-decoration: none;
```
С такими вещами нужно быть аккуратнее.
Возможно, когда-нибудь появится необходимость добавить еще ссылок, которых нет в макете, и пользователь даже не поймет, что это ссылка.

petitechatte replied:
Но в целом это не ошибка. Просто информация к размышлению.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.main-navigation__link:hover,
+.main-navigation__link:focus {
+	opacity: 0.8;
```
На главной странице логотип не должен реагировать на наведение, т.к. ссылка неактивна.

Это можно решить либо через каскад: 
.inner-page .main-navigation__link:hover,

либо воспользоваться наличием атрибута href в активных ссылках:
a[href].main-navigation__link:hover

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+.main-navigation__link:active {
+	opacity: 0.3;
```
То же самое: правило не должно работать на главной странице.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.main-header {
+	background-color: #eeeeee;
+	background-image: url(../img/background-header.png);
```
Фоновое изображение здесь лишнее.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+.main-navigation {
+	font-size: 16px;
```
Размер шрифта можно не указывать, так как он наследуется от `body`.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.main-navigation a {
+	color: #000000;
+	text-transform: uppercase;
```
text-transform ссылки наследуют по умолчанию, дублировать не надо. Переопределяется только цвет.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+/* ?Корзина эмблема */
+
+.main-navigation__basket-link a{
+	background-image: url(../img/cart-icon.png);
```
Адреса желательно заворачивать в кавычки.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.slide__lead {
+	font-size: 55px;
+	line-height: 55px;
+	font-weight: 500;
```
Заголовок в макете черного цвета!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	background-color: #fb565a;
+	text-transform: uppercase;
+	border-radius: 3px;
+}
```
Во-первых, выровняй код.
Во-вторых, разбей эти правила на два селектора. Общие стили пропиши в .button, остальное - в .slide__link.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.slide__link:hover,
+.slide__link:focus {
+	background-color: #e74246;
```
Это поведение повторяется для большинства кнопок, его стоит задать для .button.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.slide__link:hover,
+.slide__link:focus {
+	background-color: #e74246;
+	color: #ffffff ;
```
Цвет текста при наведении не меняется, нет смысла дублировать эту строку.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.slide__link:active {
+	background-color: #d7373b;
+	box-shadow: inset 0px 3px 0px 0px rgba(0,1,1,0.09);
+	color: #ffffff;
```
Это поведение повторяется и в других кнопках, можно прописать его для .button.
Цвет текста попробуй тоже сделать через rgba.

petitechatte replied:
Между цифрами в rgba ставь пробелы.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	box-shadow: inset 0px 3px 0px 0px rgba(0,1,1,0.09);
+	color: #ffffff;
+}
+.slide__link a:active{
```
Ошибка! .slide__link - это и есть ссылка, поэтому твой код не работает. Потыкай сам в кнопку.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+.slide-controls__button {
+	background-color: #eeeeee;
```
Эти кружочки чисто белого цвета, потыкай внимательнее.

petitechatte replied:
Также можно сразу убрать у них рамки и скруглить.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.slide-controls__button:hover,
+.slide-controls__button:focus {
+	background-color: #dfdfdf;
```
Поведение переключателей в стайлгайде не прописано, но почему бы и нет? Ты молодец!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.slide-controls__button:active {
+	background-color: #d5d5d5;
+	box-shadow: 0 0 10px rgba(0, 0, 0, 0.2)
```
Забыл точку с запятой в конце.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.advantages__item h3 {
+    font-size: 24px;
+	line-height: 30px;
```
Поправь выравнивание кода.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	background-position: left top;
+	background-repeat: no-repeat;
+}
+
```
У всех фоновых изображений на слайдах одинаковые свойства, за исключением адреса картинки. Это надо использовать.
Все одинаковые свойства прописываются для класса .item, а адрес - для классов .item-1/2/3. В этом и есть суть подхода элемент/модификатор. Кстати, по БЭМ модификаторы обозначаются двойным дефисом.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+    font-weight: 500;
+	background-color: #fb565a;
+	color: #ffffff;
+}
```
Выровняй код.

petitechatte replied:
Здесь часть свойств можно прописать для глобального класса .button, часть для элемента .advantages__button, а для модификаторов .advantages__button--2 .advantages__button--3 останется записать только цвет фона.
Используй каскад! CSS - каскадная таблица стилей.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	color: #ffffff;
+	box-shadow: inset 0px 3px 0px 0px rgba(0,1,1,0.09);
+}
+
```
Поведение при наведении и нажатии лучше прописать для глобального класса .button. Код сильно сократиться.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	box-shadow: inset 0px 3px 0px 0px rgba(0,1,1,0.09);
+}
+
+.advantages__button1 a:active {
```
Неправильный селектор!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.advantages__button1 a:active {
+	opacity: 0.3;
+}
+
```
Дальше все кнопки нужно переделать. Для них прописывается только цвет.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	line-height: 30px;
+	text-transform: uppercase;
+}
+
```
Этот селектор уже был выше.
Размещай селекторы по порядку, будешь меньше путаться и не будет дубликатов.

petitechatte replied:
Цвет текста черный!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+.studio__information {
+	background-image: url(../img/nerds-illustration.png);
```
Адрес лучше заворачивать в кавычки.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.information__percents {
+	font-size: 45px;
+	line-height: 10.06px;
```
Line-height тут какой-то очень странный, скорее всего ошибка дизайнера. Должен быть не меньше размера шрифта.

petitechatte replied:
Посмотри, как выглядит знак % на макете. Попробуй добавить `sup` в разметку или использовать другие способы стилизовать его.

petitechatte replied:
Текст черного цвета!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.studio__lead {
+	font-size: 45px;
+	line-height: 45px;
+	font-weight: 500;
```
Цвет текста черный!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+.studio__title-items {
+	text-transform: uppercase;
```
Текст черного цвета!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.studio__information span {
+	text-transform: uppercase;
+	font-weight: 700;
```
Текст черного цвета!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.information__descriptions {
+	font-size: 16px;
+	line-height: 18px;
+	font-weight: 400;
```
Размер шрифта и жирность наследуется от `body`. Имеет смысл указывать только межстрочный интервал.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.main-footer__contacts {
+	background-color: #ffffff;
+}
```
Цвет футера указывать не нужно, он наследуется от `body`.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	font-size: 18px;
+	line-height: 30px;
+	text-transform: uppercase;
+	font-weight: 700;
```
Текст черного цвета!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	line-height: 18px;
+	font-weight: 400;
+}
+
```
В контактах текст светло-серого цвета! Его можно прописать для `p`, но лучше для всего блока .contacts__contacts.

petitechatte replied:
Кстати, размер и жирность текста наследуются от body. Ты их для того в body и указываешь, чтобы потом не прописывать для каждого элемента.
Интерлиньяж 18px тоже часто встречается, можно завести для такого сжатого текста отдельный класс.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	color: #ffffff;
+    box-shadow: inset 0px 3px 0px 0px rgba(0,1,1,0.09);
+}
+
```
Все эти стили повторяются, они должны быть в глобальном классе .button.

petitechatte replied:
Цвет текста при нажатии должен меняться, как и в других таких же кнопках.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	display: block;
+	width: 1440px;
+	height: 416px;
+	background-image: url(../img/map-image.png);
```
Адрес следует обернуть в кавычки.
Перед background-image всегда прописывается подходящий background-color!

petitechatte replied:
С картой здесь проблема.
Карта - ценное контентное изображение, добавлять его фоном нежелательно. Но здесь допускается такой костыль для упрощения верстки.
Но в идеале надо делать все-таки тегом img и возиться с позиционированием.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	background-position: top center;
+	background-repeat: no-repeat;
+}
+
```
Карта у тебя не влазит на сайт, потому что лежит в центровщике, который уже карты.
Тут нужно либо менять разметку и доставать карту из центровщика, либо делать отрицательные margin.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	display: block;
+	width: 231px;
+	height: 190px;
+	background-image: url(../img/map-marker.png);
```
Адрес следует обернуть в кавычки.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	height: 190px;
+	background-image: url(../img/map-marker.png);
+	background-size: 231px 190px;
+	background-position: top center;
```
Положение маркера нужно указать как на макете.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.social__social-links a {
+	color: #000000;
+	text-transform: uppercase;
+}
```
Большого смысла в этих стилях не вижу, так как текст на сайте все равно не будет виден.

petitechatte replied:
Кстати этот комбинированный селектор выбирает те же ссылки, что и селектор ниже. Выбери какой-то один способ записи селектора.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	width: 79px;
+    height: 79px;
+	background-color: #e1e1e1;
+}
```
Нужно подравнять код.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.social__link:active {
+	background-color: #d7373b;
+	box-shadow: inset 0px 3px 0px 0px rgba(0,1,1,0.09);
```
Ставь пробелы между цифрами в rgba.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	box-shadow: inset 0px 3px 0px 0px rgba(0,1,1,0.09);
+}
+
+.social__link a:active {
```
Здесь опять неправильный селектор. Внутри ссылки нет и не может быть другой ссылки.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.main-footer__social b {
+	font-size: 36px;
+	line-height: 36px;
+	font-weight: 700;
```
Текст черного цвета!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	line-height: 36px;
+	font-weight: 700;
+}
+
```
Не указан цвет текста для предложения скидки!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	font-size: 55px;
+	line-height: 55px;
+	font-weight: 500;
+	text-align: center;
```
Текст черного цвета!

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+.form__slider{
+	font-size: 16px;
```
Размер шрифта наследуется от `body`, указывать необязательно.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.form__slider input {
+	background-color: #eeeeee;
+}
```
Это так не работает, к сожалению.
Посмотри, код разметки, который я тебе предложила. Фоновый цвет нужно прописать для блока-обертки .cost__slider.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.form__slider .form__label{
+	text-transform: uppercase;
+	background-color: #eeeeee;
```
Тут ты все перепутал. Фоновый цвет нужен не лейблам, а аутпутам.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+.filter .form__radio ,
+.filter .form__checkbox ,
```
Чекбоксы и радиокнопки стилизуются очень плохо, их придется прятать.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.filter .form__radio:hover ,
+.filter .form__radio:focus ,
+.filter .form__checkbox:hover ,
+.filter .form__checkbox:focus ,
```
Эти все свойства окажутся не нужны.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.filter .form__checkbox:hover ,
+.filter .form__checkbox:focus ,
+.filter .form__label:hover ,
+.filter .form__label:focus {
```
Лейбл не может быть в фокусе!
Вместо этого можно использовать комбинированный селектор:
.filter .form__radio:focus + form__label
.filter .form__checkbox:focus + form__label

В принципе, все чекбоксы и радиокнопки на сайте скорее всего подразумеваются одинаковыми, так что .filter можно вообще не указывать.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.filter .form__radio:disabled ,
+.filter .form__checkbox:disabled ,
+.form__label:disabled {
```
То же самое: label не может быть disabled. Нужно использовать комбинированные селекторы, как указано выше.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+}
+
+.form__form-button:active {
+	background-color: #d5d5d5;
```
У этой кнопки тоже добавляется внутренняя тень.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	font-size: 14px;
+	line-height: 18px;
+	font-weight: 400;
+	opacity: 0.3;
```
Размер текста и жирность наследуются от `body`.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+
+.layouts-sort__item button {
+    background-color: #000000;
+	opacity: 0.3;
```
По макету здесь цвет #a6a6a6 и 20% непрозрачность (смотри стайлгайд).

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	line-height: 18px;
+	font-weight: 400;
+	opacity: 0.3;
+}
```
Необходимо добавить поведение ссылок по стайлгайду.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+    background-color: #000000;
+	opacity: 0.3;
+}
+
```
Здесь алгоритм такой:
1. Задаем серый цвет треугольникам. Opacity 20%.
2. Переопределяем черный цвет для текущего треугольника. Opacity 100%.
3. Добавляем фокус и наведение - opacity 100% (на текущую ссылку это никак влиять не будет).

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.gallery__hover {
+    background-color: #eeeeee;
+	text-align: center;
+}
```
Нужно выровнять код.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	font-size: 18px;
+	line-height: 30px;
+    font-weight: 700;
+}
```
Нужно выровнять код.
Таких заголовков на сайте много, лучше сделать один класс для всех.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	font-size: 16px;
+	line-height: 18px;
+	font-weight: 500;
+}
```
Нужно выровнять код.

petitechatte replied:
Судя по стайлгайду, цена - это ссылка, которая ведет себя практически как базовая кнопка (глобальный класс .button).
Отличием является только отсутствие полупрозрачности текста при нажатии. Можно прописать для этого дополнительный стиль. А можно посчитать, что дизайнер просто забыл доделать полупрозрачность, и оставить все стили такими же. Спросить-то не у кого...

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	color: #666666;
+	font-size: 16px;
+	line-height: 18px;
+	font-weight: 400;
```
Размер шрифта и жирность наследуются от `body`. Для часто встречающегося правила line-height: 18px; можно сделать отдельный класс. Так что здесь можно оставить только правило для цвета.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.pagination__button {
+	background-color: #eeeeee;
+	border: none;
+	text-transform: uppercase;
```
Нужно переделать кнопки в ссылки и явно прописать им черный цвет.
Про границу. Это не очевидно, но ее лучше сделать, причем такого же цвета, что и фон. Для текущей страницы, меняется цвет фона и границы, но размер останется таким же.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+.pagination__item--current button{
+    background-color: #ffffff;
+	border: 3px solid #dbdbdb;
+}
```
Нужно выровнять код.

petitechatte commented on css/style_index.css:
```diff
@@ -0,0 +1,935 @@
+	border: 3px solid #dbdbdb;
+}
+
+/* !Сеточные стили */
```
Сетки пока не проверяю.

petitechatte reviewed (changes requested):
Проверила все, кроме сеток.
Нужно удалить все лишние картинки, добавить модальное окно, поправить скрытые слайды - добавить им фоновые изображения и сделать все слоганы в 2 строки.
Ошибок очень много.

Nomad48JMS committed:
Перенес данные.

Nomad48JMS committed:
Часть исправления ошибок.

Часть исправления ошибок.

Nomad48JMS committed:
Исправлял ошибки по 3 пулреквесту.

Исправлял ошибки по 3 пулреквесту.

Nomad48JMS committed:
Завершил иправление верстки.

Завершил иправление верстки.
